relay-system/src/service.rs

```diff
@@ -1043,7 +1043,7 @@ impl ServiceRunner {
 
     /// Starts a service and starts tracking its join handle, given a predefined receiver.
     pub fn start_with<S: Service>(&mut self, service: S, rx: Receiver<S::Interface>) {
-        self.0.push(spawn!(service.run(rx)));
+        self.0.push(spawn(S::name(), service.run(rx)));
```

Comment:
Stupid idea, we could have `spawn(TaskId::for_service::<S>())` and remove the `From<str>` impl, meaning you can actually only use it for services, everything else needs to go through the macro?